pokedex/Paging.js

```diff
@@ -0,0 +1,77 @@
+import Component from '../Component.js';
+
+class Paging extends Component {
+
+    onRender(dom) {
+        const prevButton = dom.querySelector('.prev');
+        const nextButton = dom.querySelector('.next');
+
+        if (!prevButton) {
+            return;
+        }
+
+        let page = 1;
+
+        function updateControls(){
+            const queryString = window.location.hash.slice(1);
+            const searchParams = new URLSearchParams(queryString);
+
+            const parsedPage = parseInt(searchParams.get('page'));
+            if (isNaN(parsedPage)) {
+                page = 1;
+            }
+            else {
+                page = parsedPage;
+            }
+        }
+
+        updateControls();
+
+        window.addEventListener('hashchange', () => {
+            updateControls();
+        });
+
+        function updatePage(increment) {
+            const queryString = window.location.hash.slice(1);
+            const searchParams = new URLSearchParams(queryString);
+            searchParams.set('page', page + increment);
+
+            window.location.hash = searchParams.toString();
+        }
+
+        prevButton.addEventListener('click', () => {
+            updatePage(-1);
+        });
+        
+        nextButton.addEventListener('click', () => {
+            updatePage(1);
+        });
+    }
+    renderHTML() {
+        const perPage = 10;
+        const count = this.props.count;
+        const queryString = window.location.hash.slice(1);
+        const searchParams = new URLSearchParams(queryString);
+
+        let page = 1;
+        const parsedPage = parseInt(searchParams.get('page'));
+        if (isNaN(parsedPage)) {
+            page = 1;
+        }
+        else {
+            page = parsedPage;
+        }
+
+
+        const lastPage = Math.ceil(count / perPage);
```

Comment:
This isn't working. you're stating 10 per page up above, but you're still rendering 20 pokemon so the last page that contains a pokemon is actually page 41 not 81